src/Rules.res

```diff
@@ -1,7 +1,3 @@
-module Validation = {
-  type t = Sync(string => bool) | Async(string => Js.Promise.t<bool>)
```

Comment:
While i do understand the fix with the union type i'm not sure i properly understand why type Validation.t failed. I used the Sync variant constructor with a string => bool function. What is the representation of this type at runtime?

Reply:
I made a [demo](https://rescript-lang.org/try?code=LYewJgrgNgpgBANQIZQJZiQF1SAdnAXjgG8AoOOTATwAd5NC4BlK3AYwAoBnTAJ1VwBzQgD44AIxAgoASjgAfOAEEurTj35DRcAFJcAdAAVeIYKi4x9mADyTpImaQC+pV3v1QQgjsjQZsePos7BwA+gQiAGYoFjKOpKCQsHDGIHS8vuhYOLgAXGRwAJDUdJRw5BRwsAyq7Llw3HwCghF2stqYFRTVcEi1bPWNmi0ibXIRlM5EBYUAAhC4kgAeMGCUtPC8MGxlREq4VBwA5EgyuZ0UXVUwNWqM3GoAYgsDcBrN2mPa+4f9z+yOS6Va4MPp3IgcMHsf6vd5aCZfCY-SF-F6OFyXUjuTzeVLpTL+HL6fphCLRKCxGRAA) to show you here @vtrofin 

So in short, using variant Constructor it will be interpreted to be an object as you can see in the image, so that is why you got the error no a function
<img width="1439" alt="Screenshot_2021_07_09_19_25" src="https://user-images.githubusercontent.com/30365498/125064294-759b9900-e0eb-11eb-8a88-aa62db7cb9db.png">

Reply:
oh cool, definitely. thank you for the clarification @infothien